### DISCUSSION TITLE
Developer workflow for i3 ?

### DISCUSSION THREAD

**Discussion by srbdev** [i3/i3#5628] [Q&A]
Hi,

I was wondering what the workflow would be for someone looking into building from source, making some modifications and then seeing these changes live. I've used these instructions (see [Debian and Ubuntu repositories](https://i3wm.org/docs/repositories.html)) to install i3 on my Ubuntu box and I'm now looking at this documentation for [Hacking i3](https://i3wm.org/docs/hacking-howto.html). It describes how to build but not really how to install or how to progress from there...  

- What's the install step?
- Is reverting, checking out a specific version, etc. and building/installing again enough to revert changes?

I'd love to hear some insight from contributors on this project. Thanks!

PS: If I missed a post on here or r/i3wm that already answers these questions, feel free to point me to it! 😄

**Comment by okraits** [comment]
I think installing i3 is done by `meson install` (as root) after the steps listed in https://i3wm.org/docs/hacking-howto.html#_building_i3

Doing these steps again should be sufficient to update an existing installation of i3 (from source).

**Comment by orestisfl** [comment]
I usually run my local copy of i3 inside xephyr to avoid affecting
system-wide files that are tracked by my package manager

**Reply by srbdev** [reply]
I like the Xephyr approach- I'll be looking into that. Thanks for the suggestion!
